pkg/cmd/openshift-tests/run-disruption/disruption.go

```diff
@@ -56,122 +61,183 @@ func NewRunInClusterDisruptionMonitorCommand(ioStreams genericclioptions.IOStrea
-		}
+func (f *RunAPIDisruptionMonitorFlags) ToOptions() (*RunAPIDisruptionMonitorOptions, error) {
+	originalOutStream := f.IOStreams.Out
+	closeFn, err := f.OutputFlags.ConfigureIOStreams(f.IOStreams, f)
+	if err != nil {
+		return nil, err
 	}
 
-	eventDir := filepath.Join(opt.ArtifactDir, monitorapi.EventDir)
-	if err := os.MkdirAll(eventDir, os.ModePerm); err != nil {
-		fmt.Printf("Failed to create monitor-events directory, err: %v\n", err)
-		return err
+	namespace, _, err := f.ConfigFlags.ToRawKubeConfigLoader().Namespace()
+	if err != nil {
+		return nil, err
+	}
+	if len(namespace) == 0 {
+		return nil, fmt.Errorf("namespace must be specified")
 	}
 
-	timeSuffix := fmt.Sprintf("_%s", time.Now().UTC().Format("20060102-150405"))
-	if err := monitorserialization.EventsToFile(filepath.Join(eventDir, fmt.Sprintf("e2e-events%s.json", timeSuffix)), intervals); err != nil {
-		fmt.Printf("Failed to write event data, err: %v\n", err)
-		return err
+	restConfig, err := f.ConfigFlags.ToRESTConfig()
+	if err != nil {
+		return nil, err
+	}
+	kubeClient, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return nil, err
 	}
-	fmt.Fprintf(opt.Out, "\nEvent data written, exiting\n")
 
-	return nil
+	return &RunAPIDisruptionMonitorOptions{
+		KubeClient:        kubeClient,
+		KubeClientConfig:  restConfig,
+		OutputFile:        f.OutputFlags.OutFile,
+		LoadBalancerType:  f.LoadBalancerType,
+		StopConfigMapName: f.StopConfigMapName,
+		Namespace:         namespace,
+		CloseFn:           closeFn,
+		OriginalOutFile:   originalOutStream,
+		IOStreams:         f.IOStreams,
+	}, nil
 }
 
-// StartAPIAvailability monitors just the cluster availability
-func StartAPIAvailability(ctx context.Context, restConfig *rest.Config, lb backend.LoadBalancerType) (monitorapi.Recorder, error) {
-	recorder := monitor.NewRecorder()
+// RunAPIDisruptionMonitorOptions sets options for api server disruption monitor
+type RunAPIDisruptionMonitorOptions struct {
+	KubeClient        kubernetes.Interface
+	KubeClientConfig  *rest.Config
+	OutputFile        string
+	LoadBalancerType  string
+	StopConfigMapName string
+	Namespace         string
+
+	OriginalOutFile io.Writer
+	CloseFn         iooptions.CloseFunc
+	genericclioptions.IOStreams
+}
 
-	client, err := kubernetes.NewForConfig(restConfig)
-	if err != nil {
-		return nil, err
+func (o *RunAPIDisruptionMonitorOptions) Run(ctx context.Context) error {
+	ctx, cancelFn := context.WithCancel(ctx)
+	defer cancelFn()
+
+	fmt.Fprintf(o.Out, "Starting up.")
+
+	startingContent, err := os.ReadFile(o.OutputFile)
+	if err != nil && !os.IsNotExist(err) {
+		return err
 	}
-	if err := controlplane.StartAPIMonitoringUsingNewBackend(ctx, recorder, restConfig, lb); err != nil {
-		return nil, err
+	if len(startingContent) > 0 {
+		// print starting content to the log so that we can simply scrape the log to find all entries at the end.
+		o.OriginalOutFile.Write(startingContent)
```

Comment:
This along with [static filenames](https://github.com/openshift/origin/blob/e831cef792e0dce27f2ce13ae34e619520e25637/pkg/monitortests/kubeapiserver/disruptioninclusterapiserver/manifests/dep-internal-lb.yaml#L32) reminds me of https://github.com/openshift/origin/pull/28613/files.  We were getting duplication of data when we switched from upgrade to conformance monitors.

Look at the disruption in the conformance intervals for https://prow.ci.openshift.org/view/gs/test-platform-results/logs/openshift-origin-28982-ci-4.17-e2e-aws-upgrade-ovn-single-node/1821885062338056192 and note the blocks that begin before the conformance runs.

We had to inject a unique deployment id to prevent the duplication.